Homework3/Task1PianoKeys/Key.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1PianoKeys
+{
+    struct Key : IComparable<Key>
+    {
+        public Notes Note;
+        public Octaves Octave;
+        public Accidentals Accidental;
+
+        public Key (Notes note, Octaves octave, Accidentals accidental = 0)
+        {
+            Note = note;
+            Octave = octave;
+            Accidental = accidental;
+
+            if(Octave == Octaves.Zero)
+            {
+                if(Note != Notes.A && Note != Notes.B)
+                {
+                    throw new ArgumentException("Octave 0 has only notes A, B");
+                }
+            }
+            else if(Octave == Octaves.Eighth)
+            {
+                if(Note != Notes.C)
+                {
+                    throw new ArgumentException("Octave 8 has only note C");
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder temp = new();
+
+            if(Accidental == 0)
+            {
+                temp.Append($"{Note}{(int)Octave}");
+            }
+            else
+            {
+                temp.Append($"{Note}{(char)Accidental}{(int)Octave}");
+
+            }
+
+            return temp.ToString();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if(!(obj is Key))
+            {
+                return false;
+            }
+            
+            var testKey = (Key)obj;
+
+            if(this.Octave == testKey.Octave && this.Note == testKey.Note && this.Accidental == testKey.Accidental)
```

Comment:
it was enough to test only Octave

Reply:
I'm checking if it's the same key. Since an octave has several different keys I don't think it would be enough to only check for octave.